hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/records/TxnAwareRecordsHistorian.java

```diff
@@ -137,6 +144,15 @@ public void saveExpirableTransactionRecords() {
         topLevelStreamObj =
                 new RecordStreamObject(topLevelRecord, accessor.getSignedTxnWrapper(), consensusNow, sidecars);
 
+        // feature flag if contracts nonces externalization is enabled
+        if (dynamicProperties.isContractsNoncesExternalizationEnabled()) {
+            if (accessor.getFunction().equals(HederaFunctionality.ContractCreate)) {
+                topLevelRecord.getContractCreateResult().setContractNonces(contractNonces);
```

Comment:
⚠️ As demonstrated [here](https://github.com/hashgraph/hedera-services/pull/7144), it is not safe to share the `contractNonces` map with `EvmFnResult` objects, because they will read its contents asynchronously on different threads.

This could be maximally catastrophic with high `ContractCall` TPS, since every node in the network would likely end up producing a different sequence of records, breaking consensus (not to mention mirror nodes, which would no longer see a consensus record stream running hash). 😱

Reply:
Resolved with the suggested approach following the pattern we use in `trackNewlyCreatedAccounts()`